src/app/services/actions/query-action-creators.ts

```diff
@@ -106,6 +117,20 @@ export function runQuery(query: IQuery) {
     );
   }
 
+  async function handleClaimsChallenge(response: Response, dispatch: Function): Promise<void>{
```

Comment:
We can have this handleClaimsChallenge function working purely without dispatching events. Its result can then determine which event to dispatch.

So that it does only one thing